builtin/providers/postgresql/resource_postgresql_database_funcs.go

```diff
@@ -0,0 +1,82 @@
+package postgresql
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/lib/pq"
+)
+
+func resourcePostgresqlDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*sql.DB)
+	dbName := d.Get("name").(string)
+	dbOwner := d.Get("owner").(string)
+	var dbOwnerCfg string
+	if dbOwner != "" {
+		dbOwnerCfg = fmt.Sprintf("WITH OWNER=%s", pq.QuoteIdentifier(dbOwner))
+	} else {
+		dbOwnerCfg = ""
+	}
+
+	query := fmt.Sprintf("CREATE DATABASE %s %s", pq.QuoteIdentifier(dbName), dbOwnerCfg)
+	_, err := conn.Query(query)
+	if err != nil {
+		return fmt.Errorf("Error creating postgresql database: %s", err)
+	}
+
+	d.SetId(dbName)
+
+	return nil
+}
+
+func resourcePostgresqlDatabaseDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*sql.DB)
+	dbName := d.Get("name").(string)
+
+	query := fmt.Sprintf("DROP DATABASE %s", pq.QuoteIdentifier(dbName))
+	_, err := conn.Query(query)
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourcePostgresqlDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*sql.DB)
+	dbName := d.Get("name").(string)
+
+	var owner string
+	err := conn.QueryRow("SELECT pg_catalog.pg_get_userbyid(d.datdba) from pg_database d WHERE datname=$1", dbName).Scan(&owner)
+	switch {
+	case err == sql.ErrNoRows:
+		d.SetId("")
+		return nil
+	case err != nil:
+		return fmt.Errorf("Error reading info about database: %s", err)
+	default:
+		d.Set("owner", owner)
```

Comment:
If we do `CREATE DATABASE` without the `WITH OWNER` clause above, and then we read here, will this get populated with a default owner?

If so, it's important to mark the `owner` argument as *both* `Optional` and `Computed`. This tells Terraform that the user can optionally override it (the `Optional` part) but if they *don't* then the value will be computed based on something the server knows.